### PR TITLE
Kses - Part 13

### PR DESCRIPTION
**Pull request by hoseinrafiei**
This is related to https://github.com/eventespresso/event-espresso-core/issues/3738 issue.

Escaped some more variables in messaging system and removed some instances from PHP comments.
Related Checklist: https://trello.com/c/Kdc3fASn/1163-ee4-core-testing-checklist-pr-3853

### CODE REVIEW

File: core/EE_Error.core.php [hunk 1]
```diff
@@ -1160,7 +1105,7 @@ public static function get_raw_notices()
      * @return void
      * @throws InvalidDataTypeException
      */
-    public static function add_persistent_admin_notice($pan_name = '', $pan_message, $force_update = false)
+    public static function add_persistent_admin_notice($pan_name, $pan_message, $force_update = false)
```
Comment (tn3rb) [inline core/EE_Error.core.php:1108]: 👍🏻

File: core/db_classes/EE_Base_Class.class.php [hunk 2]
```diff
@@ -2745,16 +2745,6 @@ public function model_field_array()
...
-     * Example: in functions.php (or in a plugin):
-     *      add_filter('FHEE__EE_Answer__my_callback','my_callback',10,3);
-     *      function my_callback($previousReturnValue,EE_Base_Class $object,$argsArray){
-     *          $returnString= "you called my_callback! and passed args:".implode(",",$argsArray);
-     *          return $previousReturnValue.$returnString;
-     *      }
-     * require('EE_Answer.class.php');
-     * $answer= EE_Answer::new_instance(array('REG_ID' => 2,'QST_ID' => 3,'ANS_value' => The answer is 42'));
-     * echo $answer->my_callback('monkeys',100);
-     * //will output "you called my_callback! and passed args:monkeys,100"
```
Comment (tn3rb) [inline core/db_classes/EE_Base_Class.class.php:2757]: why did you remove this example? 
can you plz restore this?

File: core/db_models/EEM_Base.model.php [hunk 3]
```diff
@@ -5585,16 +5585,7 @@ public function item_name($quantity = 1)
...
-     * returns will be returned by this function. Example: in functions.php (or in a plugin):
-     * add_filter('FHEE__EE_Answer__my_callback','my_callback',10,3); function
-     * my_callback($previousReturnValue,EE_TempBase $object,$argsArray){
-     * $returnString= "you called my_callback! and passed args:".implode(",",$argsArray);
-     *        return $previousReturnValue.$returnString;
-     * }
-     * require('EEM_Answer.model.php');
-     * $answer=EEM_Answer::instance();
-     * echo $answer->my_callback('monkeys',100);
-     * //will output "you called my_callback! and passed args:monkeys,100"
```
Comment (tn3rb) [inline core/db_models/EEM_Base.model.php:5597]: plz restore this example as well.
if your reason was because of the `echo $` then you can simply get rid of that var by changing this:

```php
     * $answer=EEM_Answer::instance();
     * echo $answer->my_callback('monkeys',100);
```

 to this:

```php
     * echo EEM_Answer::instance()->my_callback('monkeys',100);
```

File: core/entities/models/JsonModelSchema.php [hunk 4]
```diff
@@ -266,9 +266,6 @@ protected function getProtectedFieldsSchema()
 
     /**
      * Allows one to just use the object as a string to get the json.
-     * eg.
-     * $json_schema = new JsonModelSchema(EEM_Event::instance(), new CalculatedModelFields);
-     * echo $json_schema; //outputs the schema as a json formatted string.
```
Comment (tn3rb) [inline core/entities/models/JsonModelSchema.php:271]: same here, maybe change to something like:

```php
     * $json_schema = new JsonModelSchema(EEM_Event::instance(), new CalculatedModelFields);
     * // converts the schema to a json formatted string.
```

File: core/libraries/form_sections/base/EE_Form_Section_Proper.form.php [hunk 5]
```diff
@@ -295,7 +295,7 @@ protected function getCachedRequest($req_data = null)
      * @param array|null $req_data should usually be post data (the default).
      *                             However, you CAN supply a different array.
      *                             Consider using set_defaults() instead however.
-     *                             (If you rendered the form in the page using echo $form_x->get_html()
+     *                             (If you rendered the form in the page using $form_x->get_html()
```
Comment (tn3rb) [inline core/libraries/form_sections/base/EE_Form_Section_Proper.form.php:298]: 👍🏻

File: core/libraries/messages/defaults/default/email_preview_wrapper.template.php [hunk 6]
```diff
@@ -1,4 +1,9 @@
...
+<?php
+
+/** @var string $main_body */
+use EventEspresso\core\services\request\sanitizers\AllowedTags;
+
+?>
 <div id="ee-message-preview-container" style="margin-top: 10px">
-    <?php echo $main_body; // already escaped ?>
+    <?php echo wp_kses($main_body, AllowedTags::getWithFormTags()); ?>
 </div>
```
Comment (tn3rb) [inline core/libraries/messages/defaults/default/email_preview_wrapper.template.php:9]: ooo surprised the linter didn't scream about missing a newline at the end of the file.
maybe that only applies to classes 🤔

File: core/libraries/messages/messenger/EE_Html_messenger.class.php [hunk 7]
```diff
@@ -473,7 +475,8 @@ protected function _send_message()
         );
         $this->_deregister_wp_hooks();
         add_action('wp_enqueue_scripts', array($this, 'enqueue_scripts_styles'));
-        echo $this->_get_main_template(); // already escaped
+
+        echo $this->_get_main_template();
```
Comment (tn3rb) [inline core/libraries/messages/messenger/EE_Html_messenger.class.php:479]: can this one not be escaped?

File: core/services/request/sanitizers/AllowedTags.php [hunk 8]
```diff
@@ -66,6 +66,16 @@ class AllowedTags
...
+        'topmargin'         => 1,
+        'leftmargin'        => 1,
+        'marginheight'      => 1,
+        'marginwidth'       => 1,
+        'property'          => 1,
+        'bgcolor'           => 1,
+        'media'             => 1,
+        'cellpadding'       => 1,
+        'cellspacing'       => 1,
+        'border'            => 1,
```
Comment (tn3rb) [inline core/services/request/sanitizers/AllowedTags.php:78]: 👍🏻

File: core/services/request/sanitizers/AllowedTags.php [hunk 9]
```diff
@@ -161,16 +176,16 @@ private static function initializeWithFormTags()
...
-            'form' => AllowedTags::$attributes,
-            'label' => AllowedTags::$attributes,
-            'input' => AllowedTags::$attributes,
-            'select' => AllowedTags::$attributes,
-            'option' => AllowedTags::$attributes,
+            'form'     => AllowedTags::$attributes,
+            'label'    => AllowedTags::$attributes,
+            'input'    => AllowedTags::$attributes,
+            'select'   => AllowedTags::$attributes,
+            'option'   => AllowedTags::$attributes,
```
Comment (tn3rb) [inline core/services/request/sanitizers/AllowedTags.php:183]: 👍🏻

File: core/services/request/sanitizers/AllowedTags.php [hunk 10]
```diff
@@ -184,12 +199,39 @@ private static function initializeWithScriptAndStyleTags()
...
+            'select'    => AllowedTags::$attributes,
+            'option'    => AllowedTags::$attributes,
+            'optgroup'  => AllowedTags::$attributes,
+            'textarea'  => AllowedTags::$attributes,
+            'button'    => AllowedTags::$attributes,
+            'fieldset'  => AllowedTags::$attributes,
+            'output'    => AllowedTags::$attributes,
+        ];
+        AllowedTags::$allowed_with_full_tags = array_merge_recursive($all_tags, $full_tags);
+    }
```
Comment (tn3rb) [inline core/services/request/sanitizers/AllowedTags.php:233]: noice 👌🏻

**Review by tn3rb — CHANGES_REQUESTED** [review]
more good stuff, but there's a few of those code examples in the comments that I would like restored

File: core/db_classes/EE_Base_Class.class.php (hunk 2 above)
Comment (hoseinrafiei) [reply core/db_classes/EE_Base_Class.class.php:2757]: Restored.

File: core/db_models/EEM_Base.model.php (hunk 3 above)
Comment (hoseinrafiei) [reply core/db_models/EEM_Base.model.php:5597]: Yes, Restored with a little change.

File: core/libraries/messages/messenger/EE_Html_messenger.class.php (hunk 7 above)
Comment (hoseinrafiei) [reply core/libraries/messages/messenger/EE_Html_messenger.class.php:479]: Since it is a full HTML page with doctype, html, meta, script, style, and other tags, we cannot escape it without destroying the required contents including but not limited to embedded JS and CSS codes that are included by any plugin using WP head and footer actions of WordPress.

**Comment by hoseinrafiei** [comment]
@tn3rb Please review it again.

**Review by tn3rb — APPROVED** [review]
Awesome, thnx

Sorry for the delay, didn't see any notification for this

**Comment by hoseinrafiei** [comment]
@Saam01 Please do the checklist of this PR.

**Comment by hoseinrafiei** [comment]
@Saam01 Any update on this?

**Comment by Saam01** [comment]
I've finished testing this PR.
